[FriedCabbageSalad] iP

FriedCabbageSalad opened the pull request:
### Count

> "Simplicity is the ultimate sophistication." - Leonardo da Vinci.

Count allows you to straighten out the various tasks going on in life.

- **Text-based**
- _Simple to understand_
- ~~Slow~~ Swift to use!

Follow these simple steps to get started
1. Download it from [here](https://github.com/FriedCabbageSalad/ip/releases/tag/A-Jar)
2. Open the jar file
3. Start keeping track of your life! 😊

Current feature set as of v0.1:
- [x] Save tasks to a txt file for sharing/storage!
- [x] Load tasks from shared txt files
- [ ] Deadlines (maybe)

If you wish to change the file path which Count saves to, here is how to do it!
`    public static void main(String[] args) {
Count c = new Count("./Count.txt");
c.start();
    }`

Under the main method, of Count.java, replace "./Count.txt" with any file path you desire!
Example: "filepath/Count.txt"

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add Level-0 requirements

Commit:
Add Level-1 requirements

Commit:
Add Level-2 requirements

Commit:
Add Level-3 requirements

Commit:
Add Level-4 requirements

Commit:
Add Level-5 requirements

Commit:
Add Level-6 requirements

Commit:
Fix for runtest.bat

Commit:
Update Count.java to follow OOP standards
Add tests for input.txt
Add expected results in EXPECTED.TXT

Commit:
Add Level-7 Requirements
Create static Sava.java to handle save logic
Edit Count.java parser for "save" function

Commit:
Add Level 8 Requirements
Change Events to inherit from Deadlines

Commit:
Update EXPECTED.txt

Commit:
Add A-MoreOOP
Add A-Packages
Refactor Count to align with OOP Increment

Commit:
Merge branch 'add-gradle-support'

Commit:
Add A-Junit requirements
Edit IncorrectFormatException message in multiWordParser to remove typo

Commit:
configure build.gradle for jar creation

Commit:
Add A-JavaDoc requirements

Commit:
Add A-CodingStandard requirements

Commit:
Add Level-9 requirements

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/count/Count.java
#	src/main/java/count/Parser.java
#	src/main/java/count/Storage.java
#	src/main/java/count/TaskList.java
#	src/main/java/count/action/Action.java
#	src/main/java/count/action/AddTask.java
#	src/main/java/count/action/Delete.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/count/Parser.java

kaajinn commented on src/main/java/count/Storage.java:
```diff
@@ -0,0 +1,100 @@
+     */
+    public ArrayList<Task> load() throws IncorrectFormatException, FileNotFoundException {
+        this.sc = new Scanner(this.file);
+        ArrayList<Task> ls = new ArrayList<>();
```
Should 'ls' be renamed to avoid ambiguity?

FriedCabbageSalad replied:
Yep! I've renamed it to listOfTasksRead to be more clear.

kaajinn commented on src/main/java/count/action/Delete.java:
```diff
@@ -0,0 +1,41 @@
+    @Override
+    public String run() throws CountException {
+        try {
+            Task curr = ls.getList().get(this.index - 1);
```
Should 'curr' be renamed to avoid ambiguity?

FriedCabbageSalad replied:
Renamed 'curr' to 'taskRemoved' to be more clear

kaajinn commented on src/test/java/count/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+            "Type 'help' to see correct formatting examples";
+    String deadlineIncorrectFormatString = "Invalid format for event or deadline!\n" +
+            "Type 'help' to see correct formatting examples";
+
```
Could the following method names be made less lengthy? Because we can already tell what exception should be thrown within the method itself.

kaajinn commented on src/test/java/count/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+
+public class ParserTest {
+    String rootFilePath = "./Count.txt";
+    String markIncorrectFormatString = "Use a number after mark/unmark/delete to specify the task targeted!\n" +
```
I like these variable names 👍, very descriptive and non-ambiguous!

kaajinn reviewed (commented):
LGTM! Only very minor nits, otherwise a fantastic job on following the code quality guidelines!

kaajinn commented on src/test/java/count/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+
+public class ParserTest {
+    String rootFilePath = "./Count.txt";
+    String markIncorrectFormatString = "Use a number after mark/unmark/delete to specify the task targeted!\n" +
+            "Type 'help' to see correct formatting examples";
+    String deadlineIncorrectFormatString = "Invalid format for event or deadline!\n" +
```
I like these variable names 👍, very descriptive and non-ambiguous!

kaajinn commented on src/test/java/count/ParserTest.java:
```diff
@@ -0,0 +1,43 @@
+    String deadlineIncorrectFormatString = "Invalid format for event or deadline!\n" +
+            "Type 'help' to see correct formatting examples";
+
+    @Test
+    public void singleWordParser_invalidCommandExceptionThrown() {
+        Parser p = new Parser(new TaskList(), rootFilePath);
+        assertThrows(InvalidCommandException.class, () -> p.parse("BLAAH"));
+    }
+
+    @Test
+    public void multiWordParser_invalidCommandExceptionThrown() {
+        Parser p = new Parser(new TaskList(), rootFilePath);
+        assertThrows(InvalidCommandException.class, () -> p.parse("BLAAH RAAAA"));
+    }
+
+    @Test
+    public void multiWordParser_mark_incorrectFormatExceptionThrown() {
+        Parser p = new Parser(new TaskList(), rootFilePath);
+        Exception exception = assertThrows(IncorrectFormatException.class, () -> p.parse("mark cheese"));
+        assertEquals(markIncorrectFormatString, exception.getMessage());
+    }
+
+    @Test
+    public void multiWordParser_deadline_incorrectFormatExceptionThrown() {
+        Parser p = new Parser(new TaskList(), rootFilePath);
+        Exception exception = assertThrows(IncorrectFormatException.class, () -> p.parse("deadline homework /by 300/10/2000"));
+        assertEquals(deadlineIncorrectFormatString, exception.getMessage());
+    }
```
Could the following method names be made less lengthy? Because we can already tell what exception should be thrown within the method itself.

kaajinn reviewed (commented):
Fixed the formatting for some of the previous comments I left for easier reading. My bad!

Commit:
Change variable names to meet Coding Quality

Commit:
Change method names to meet Coding Quality

Commit:
Add A-CheckStyle requirements
Made changes to meet checkstyle.xml

Commit:
Add Level-10 requirements

Shux347 commented on src/main/java/count/Count.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Constructor for Count object
+     * @param filePath String filePath in which Count will read and save to
+     */
```
- In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... etc. (not Return or Returning etc)
- Space between method summary and list of parameters (not explicitly required but shown in example given in java coding standards page, and also looks nice)
```suggestion
    /**
     * Constructs Count object
     *
     * @param filePath String filePath in which Count will read and save to
     */
```

Shux347 commented on src/main/java/count/Count.java:
```diff
@@ -0,0 +1,85 @@
+ * @author Kieran Koh Jun Wei
+ */
+public class Count {
+    private TaskList ls;
```
taskList would be a better name as ls is not immediately recognised

pradyuprasad replied:
I second the other reviewer's comment that ls is not descriptive of what this does. Perhaps you could call it `taskList`

Shux347 commented on src/main/java/count/action/Help.java:
```diff
@@ -0,0 +1,26 @@
+package count.action;
+
+/**
+ * The Greet class is used to give a message on how to format commands
```
Forgot to change name in comment?

Shux347 commented on src/main/java/count/action/Delete.java:
```diff
@@ -0,0 +1,42 @@
+
+
+/**
+ * The Delete class is used to add tasks to the TaskList
```
Forgot to change class description?

Shux347 commented on src/main/java/count/action/ListReply.java:
```diff
@@ -0,0 +1,35 @@
+import count.TaskList;
+
+/**
+ * The Delete class is used to list all the Tasks in the TaskList
```
Forgot to change class name?

Shux347 reviewed (commented):
Header comments summaries need to be changed to follow format required in our Java coding standard

You forgot to change some parts of your class description

Otherwise LGTM, great job!

pradyuprasad commented on src/main/java/count/TaskList.java:
```diff
@@ -0,0 +1,33 @@
+        this.ls = ls;
+    }
+
+    public ArrayList<Task> getList() {
```
perhaps getTaskList would be a better name to name this? It would convey the intent of the method more

pradyuprasad commented on src/main/java/count/action/Deactivate.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * The Deactivate class is used to turn the Count program off
+ */
+public class Deactivate extends Action {
```
these classes are well named and structured

pradyuprasad reviewed (commented):
Overall, this was a good job done. Nearly all standard have been followed, and the code is clear and well structured

Commit:
Add A-Assertions requirement

There is currently no check if Count is still on before GUI responses

Added an assert statement to check for it within getResponse

Commit:
Add A-CodeQuality requirements

UI.java was leftover after changing to JavaFX for GUI

Refactor Count.java to no longer use UI.java

Let's implement closing of the application window when off

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Revert "Merge branch 'branch-A-CodeQuality'"

This reverts commit a9279478d824f78adb5829f4f56a3b3eec0e1bff, reversing
changes made to 7420a935becf0529d3a4ba4c1a0d3737462d19e3.

Commit:
Revert "Add A-Assertions requirement"

This reverts commit 7420a935becf0529d3a4ba4c1a0d3737462d19e3.

Commit:
Add A-Assertions requirement

There is currently no check if Count is still on before GUI responses

Added an assert statement to check for it within getResponse

Commit:
Add A-CodeQuality requirements

UI.java was leftover after changing to JavaFX for GUI

Refactor Count.java to no longer use UI.java

Let's implement closing of the application window when off

FriedCabbageSalad committed:
Merge pull request #2 from FriedCabbageSalad/branch-A-Assertions

Add A-Assertions requirement

FriedCabbageSalad committed:
Merge pull request #3 from FriedCabbageSalad/branch-A-CodeQuality

Add A-CodeQuality requirements

Commit:
Add B-Reminder requirements

Commit:
Add A-MoreErrorHandling requirements

There was no checks for the Event class to ensure that the end time is strictly after the start time

InvalidTimelineException has been created to be thrown if the end time is found
to be before or equal to the start time

Count and Storage have been updated to account for these new exceptions

Commit:
add A-MoreTesting requirements

There were no JUnit tests for the ToDo, Deadline and Event classes

ParserTest is updated to account for remind

Commit:
Add A-Personality requirements

Add personality through frog sounds in text responses

Add special text responses for the following cases,
* list is empty for find, remind, list and save
* mark a complete task
* unmark an incomplete task

Commit:
Add title for jar application

Title was previously missing

Add name to introduction greeting

Commit:
Add A-Release requirements

Readme was outdated

Updated readme with non-trivial features and documentation

Add ui.png
Change to remind response message string to increase clarity

Commit:
Rename image file from ui.png to Ui.png

Updated the file name for clarity and consistency with project website naming conventions.

Commit:
Change JavaDocs to follow Java Coding Standard

Previously some Header comments summaries did not follow coding standard

Change first sentence to start in the form Returns ..., Sends ..., Adds ... etc.
Update class description to match the correct methods
Change Count.java variable name from ls to taskList for clarity reasons

FriedCabbageSalad committed:
Update README.md

There was a formatting error in README.md

Change reference to Ui.png instead of ui.png

FriedCabbageSalad committed:
Update README.md

Add instructions on installing and running Count

FriedCabbageSalad committed:
Update README.md

Add newline formatting to README.md

FriedCabbageSalad committed:
Update README.md

Add instructions for using Count

FriedCabbageSalad committed:
Update README.md

Add instructions to install and run Count with CLI

FriedCabbageSalad committed:
Update README.md

Fix heading format issues

Commit:
Add auto save after adding or deleting tasks

Commit:
Revert "Add auto save after adding or deleting tasks"

This reverts commit a2f8100bbbfa629df8fcd778a3738a801ed436b8.

FriedCabbageSalad committed:
Update README.md

Commit:
Add auto save for commands that modify task list

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Update README.md

Commit:
Update README.md
